RRT_star_unbounded.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib import patches
+from pgraph import UGraph, DGraph, UVertex
+import time
+from planning import RRT, Workspace, Rectangle, Circle
+from RRT_star import RRT_star
+import math
+
+class RRT_star_unbounded(RRT_star):
+
+    def new_point_routine(self, q, v_nearest,dist,max_edge_len,min_edge_len):
+        n =self.graph.n
+        #print(2 * math.sqrt((math.log(n) / n)))
+        #rn = min(100/n, 5) # change rn depending on how many nodes there are, look at formula on RTT star paper
+        rn = 2*max_edge_len
+        neighborhood = self.near_vertices(q,rn)
+        if len(neighborhood) == 0:
+            return
+        cost = self.cost(v_nearest) + self.connection(v_nearest.coord, q)
+        for vertex in neighborhood:
+            point = vertex.coord
+            if not self.workspace.edge_is_in_collision(q,point):
+                c_prime = self.cost(vertex) + self.connection(point, q)
+                if cost>c_prime:
+                    cost=c_prime
+                    v_nearest = vertex
+                    dist = self.connection(point, q)
+
+        # split distance into multiple edges
+        if dist > max_edge_len:
+            q_closest = v_nearest.coord
+            vertices = round(dist / max_edge_len)
+            for i in range(vertices - 1):
+                cur_vertex = q_closest + (i + 1) * (q - q_closest) / np.linalg.norm(
+                    q - q_closest
+                )
+                v_nearest, dist = self.closest_vertex(cur_vertex)
+                if dist<min_edge_len:
+                    continue
+                v = self.new_point_routine(cur_vertex,v_nearest,max_edge_len,max_edge_len,min_edge_len)
+                v_nearest = v
+
+        v = self.graph.add_vertex(q)
+        edge = v.connect(v_nearest)
+        v.parent = v_nearest
+        v.parent = v_nearest
+        v.edge = edge
+        q_min = v_nearest.coord
+
+        for vertex in neighborhood:
+            point = vertex.coord
+            dist = np.linalg.norm(point -q_min)
+            if dist== 0:
```

Comment:
Don't check for equality with floating point numbers: do something like `dist <= 1e-8`. We need a threshold due to floating point rounding error.